### ISSUE TITLE
Limit derivePointedRevCommit calls

### ISSUE THREAD

**Issue by mkondratek** [VirtusLab/git-machete-intellij-plugin#281]
I can see other places where `derivePointedRevCommit` is invoked... like in `deriveCommitsUntil`; but I don't think it should be called anywhere but just in this very line. `getPointedCommit` should be used instead to avoid double derivation. Also, for the same reason, `derivePointedRevCommit` can probably be private now.

_Originally posted by @PawelLipski in https://github.com/VirtusLab/git-machete-intellij-plugin/pull/245_

**Comment by mkondratek** [comment]
The problem is more complex than it seems. 

`RevCommit` that needs to be accessed via `GitCoreCommit#getJgitCommit` manifests mutation issues.
Reused commit gains additional flag `org.eclipse.jgit.revwalk.RevWalk#SEEN`. Moreover, the `org.eclipse.jgit.revwalk.RevCommit#reset` docs say:
```
         /**
	 * Reset this commit to allow another RevWalk with the same instances.
	 * <p>
	 * Subclasses <b>must</b> call <code>super.reset()</code> to ensure the
	 * basic information can be correctly cleared out.
	 */
	public void reset() {
		inDegree = 0;
	}
```

**Comment by PawelLipski** [comment]
Resolved by #282
